sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerClient.java

```diff
@@ -1161,19 +1155,9 @@ public PagedIterable<BlobItem> listBlobsByHierarchy(String directory) {
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedIterable<BlobItem> listBlobsByHierarchy(String delimiter, ListBlobsOptions options, Duration timeout) {
         BiFunction<String, Integer, PagedResponse<BlobItem>> func = (marker, pageSize) -> {
-            ListBlobsOptions finalOptions;
+            ListBlobsOptions finalOptions = options == null ? new ListBlobsOptions() : options;
             if (pageSize != null) {
-                if (options == null) {
-                    finalOptions = new ListBlobsOptions().setMaxResultsPerPage(pageSize);
-                } else {
-                    // Note that this prefers the value passed to .byPage(int) over the value on the options
-                    finalOptions = new ListBlobsOptions()
-                        .setMaxResultsPerPage(pageSize)
-                        .setPrefix(options.getPrefix())
-                        .setDetails(options.getDetails());
-                }
-            } else {
-                finalOptions = options == null ? new ListBlobsOptions() : options;
+                finalOptions.setMaxResultsPerPage(pageSize);
```

Comment:
So for both `listBlobsByHierarchy` and `listBlobs` now, it looks like we could be mutating the user-provided `options` instance in if `pageSize` is provided?

Looking over the async version, the implementation does not appear to have the same behavior. Would it be possible to update it to closer to this pattern that we've been using? https://github.com/Azure/azure-sdk-for-java/blob/6cc9752cf94893d6c7900243a1edbe84ba4d0c03/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClient.java#L434-L435 The big difference is that we never mutate the provided options bag. Instead, we use the explicit pageSize provided and fallback to the value retrieved from `getMaxResultsPerPage` whether the method created it or it was provided by the user. That seems like more the behavior we'll want.